File: p2.py
#Time: O(n^2)
#Space: O(n)
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        if people is None or len(people) < 1:
            return None

        people.sort(key = lambda x: (-x[0], x[1]))
        result = [None for _ in range(len(people))]

        for h, k in people:
            if result[k] is not None:
                shift_ind = len(people) - 1
                while shift_ind > k:
                    result[shift_ind] = result[shift_ind - 1]
                    shift_ind -= 1
            result[k] = [h, k]

        return result

File: p2.java
// Time Complexity : O(n)
// Space Complexity :O(n)
// Did this code successfully run on Leetcode :yes
// Any problem you faced while coding this :no


// Your code here along with comments explaining your approach

class Solution {
    public int[][] reconstructQueue(int[][] people) {
//Sort based on the height
        Arrays.sort(people, (a,b) -> {
            if(a[0] == b[0]){
                return a[1] - b[1];
            }
            return b[0] - a[0];
        });
        
        //Add to the list based on the number of people in front
        ArrayList<int[]> list = new ArrayList<>();
        for(int[] p: people){   
            list.add(p[1], p);
        }

//convert list to array
        int[][] result = new int[people.length][2];
        int i=0;
        for(int[] p: list){   
            result[i] = p;
            i++;
        }
        return result;
    }
}